### PR TITLE
[Feature] Refactor categorical dists: Masked one-hot and pass-through gradients

### PR DESCRIPTION
**Pull request by vmoens**
## Description

Implements a masked one hot distribution.
Enables 2 reparam strategies for one-hot samples: RelaxedOneHot or Pass-through

Also renamed "mask" in "action_mask" in the MaskedAction transform.

@matteobettini @MateuszGuzek

### CODE REVIEW

File: torchrl/modules/distributions/discrete.py [hunk 1]
```diff
@@ -245,3 +291,153 @@ def _mask_logits(
...
+    Keyword Args:
+        mask (torch.Tensor): A boolean mask of the same shape as ``logits``/``probs``
+            where ``False`` entries are the ones to be masked. Alternatively,
+            if ``sparse_mask`` is True, it represents the list of valid indices
+            in the distribution. Exclusive with ``indices``.
+        indices (torch.Tensor): A dense index tensor representing which actions
+            must be taken into account. Exclusive with ``mask``.
+        neg_inf (float, optional): The log-probability value allocated to
+            invalid (out-of-mask) indices. Defaults to -inf.
+        padding_value: The padding value in the then mask tensor when
```
Comment (MateuszGuzek) [inline torchrl/modules/distributions/discrete.py:317]:
```suggestion
        padding_value: The padding value in the mask tensor. When
```

**Review by MateuszGuzek — APPROVED** [review]
LGTM, single comment suggesting improvement in the docstring
